CHANGELOG.md

```diff
@@ -1,6 +1,81 @@
 Changelog
 =========
 
+## Version 4.0
+
+* Kompatibilität zu Twig 3
+* BC-Breaks:
+  * Entfallener Twig-Tag: `navigation_theme` (z.B. `{% navigation_theme root "AppBundle:Navigation:navigation.html.twig" %}`).
+    Man kann im Projekt stattdessen direkt `WebfactoryNavigationBundle:Navigation:navigation.html.twig` extenden und
+    blockweise überschreiben. Beispiel:
+      
+    ```
+    {# Datei: src/AppBundle/Resources/views/Navigation/navigation.html.twig #}
+    
+    {% extends "WebfactoryNavigationBundle:Navigation:navigation.html.twig" %}
+        
+    {% block navigation_list %}
+      <nav class="projektspezifische-wrapperklasse">
+        {{ parent() }}
+      </nav>
+    {% endblock %}
+    
+    {%- block navigation_caption -%}
+        {{ node.caption |upper }}
+    {%- endblock -%}
+    ```
+
+  * Entfallene Twig-Funktionen: `navigation_list`, `navigation_list_class`, `navigation_item`, `navigation_item_class`,
+    `navigation_text`, `navigation_text_class`, `navigation_url`, `navigation_caption`. Aufrufe dieser
+    Wrapper-Funktionen wie z.B. `{{ navigation_caption(themeRoot, node, level) }}` durch
+    `{{ block('navigation_caption') }}` (ohne Parameter) ersetzen.
+     
+     Ersatzlos entfallen: `power_set`, `navigation`.
+
+  * Entfallene Services: `webfactory_navigation.twig_theme_extension`, `Webfactory\Bundle\NavigationBundle\Twig\NavigationThemeExtension`
+    (waren vermutlich nur intern genutzt).
+
+  * Entfallender Controller: NavigationController. Aufrufe wie folgt ersetzen:
+  
+    alt:
+    ```
+    {{ render(
+      controller(
+        'WebfactoryNavigationBundle:Navigation:tree',
+        {
+          root: {"webfactory_pages.page_id": root_page_id}
+        }
+      )
+    ) }}
+    ```
+    
+    neu: `{{ navigation_tree({"webfactory_pages.page_id": root_page_id}) }}` (für weitere Parameter siehe \Webfactory\Bundle\NavigationBundle\Twig\NavigationExtension::renderTree)
+    
+    alt:
+    ```
+    {{ render(
+      controller(
+        'WebfactoryNavigationBundle:Navigation:ancestry',
+        {
+          startLevel: 1
+        }
+      )
+    ) }}
+    ```
+    
+    neu: `{{ navigation_ancestry(1) }}` (für weitere Parameter siehe \Webfactory\Bundle\NavigationBundle\Twig\NavigationExtension::renderAncestry)
```

Comment:
Ich frage mich, ob man das hier sprechender gestalten könnte?

```twig
{{ navigation_ancestry({ startLevel: 1, expandedLevels: 3 }) }}
```

Reply:
Funktioniert out of the box mit named parameters 😃 (ohne die Array/Objekt-Syntax `{}`) :

```
{{ navigation_ancestry(startLevel = 1, template = 'AppBundle:Navigation:secondaryNav.html.twig') }}
```

README und CHANGELOG sind jetzt angepasst, um diese Schreibweise zu empfehlen.